WordPress/gutenberg
# Update the document title in the template mode of the post editor

Related #50378 

## What?

This PR updates the design of the "title bar" in the "template mode" of the post editor to match the new design in the site editor. 

I considered whether I should be showing the title bar in the "regular mode" of the post editor too with the current post/page but because of the implications on the "top toolbar", I decided to leave this for a separate PR. Let me know what you think here?

## Testing Instructions

1- Open the post editor
2- Click on the "template -> edit" from the document sidebar.
3- Notice the new design of the "title bar".
